interfaces/vault_kv/v0/README.md

```diff
@@ -0,0 +1,65 @@
+# `vault-kv`
+
+## Usage
+
+Some charms require a secure key value store. This relation interface describes the expected behavior of any charm claiming to interact with Vault Key Value stores.
+
+## Direction
+
+```mermaid
+flowchart TD
+    Requirer -- mount_suffix, egress_subnet --> Provider
+    Provider -- vault_url, mount, credentials --> Requirer
+```
+
+## Behavior
+
+Both the Requirer and the Provider need to adhere to criteria to be considered compatible with the interface.
+
+### Provider
+
+Provider expectations
+
+- Must provide the vault url
+- Must provide a key value mount, the mount name shall respect the following pattern: charm-<requirer app>-<user provided suffix>
+- Must provide a role_id and role_secret_id for each unit, with access protected by unit's egress_subnet
```

Comment:
Can you extend a bit on how the egress subnet is used? That would be used when creating the approle policy right?

Reply:
The egress_subnet is used when creating an approle and when generating a secret id